modules/administration-guide/attachments/restricted-environment/prepare-restricted-environment.sh

```diff
@@ -36,26 +36,36 @@ declare prod_operator_version="${prod_operator_version:?Define the variable}"
 
 # Destination registry
 declare my_registry="${my_registry:?Define the variable}"
-declare my_catalog="${my_catalog:-restricted-environment-install}"
+declare my_catalog=restricted-environment-install
 declare my_operator_index="$my_registry/$my_catalog/my-operator-index:v${ocp_ver}"
 
 # Create local directories
 mkdir -p "${my_catalog}/${devworkspace_operator_package_name}" "${my_catalog}/${prod_operator_package_name}"
 
 echo "Fetching metadata for the ${devworkspace_operator_package_name} operator catalog channel, packages, and bundles."
 opm render "${devworkspace_operator_index}" \
-  | jq "select(\
-    .name == \"${devworkspace_operator_package_name}.${devworkspace_operator_version}\" \
-    and .schema == \"olm.bundle\" \
+  | jq "select \
+    (\
+      (.schema == \"olm.bundle\" and .name == \"${devworkspace_operator_package_name}.${devworkspace_operator_version}\") or \
+      (.schema == \"olm.package\" and .name == \"${devworkspace_operator_package_name}\") or \
+      (.schema == \"olm.channel\" and .package == \"${devworkspace_operator_package_name}\") \
     )" \
+  | jq "select \
+     (.schema == \"olm.channel\" and .package == \"${devworkspace_operator_package_name}\").entries \
+      |= [{name: \"${devworkspace_operator_package_name}.${devworkspace_operator_version}\"}]" \
```

Comment:
Tested this and it does what we expect (I think). However, this approach -- especially modifying the channel to only have one entry -- will result in catalogs that don't support upgrades, with the only flow for updating the installed version being to completely uninstall and reinstall. If this is the intended set up, feel free to ignore this comment.